filters/apiusagemonitoring/spec.go

```diff
@@ -292,56 +290,23 @@ func (s *apiUsageMonitoringSpec) buildClientTrackingInfo(apiIndex int, api *apiC
 
 // generateRegExpStringForPathTemplate normalizes the given path template and
 // creates a regular expression from it.
-func generateRegExpStringForPathTemplate(pathTemplate string) (normalizedPathTemplate, matcher, pathLabel string) {
-	pathParts := strings.Split(pathTemplate, "/")
-	matcherPathParts := make([]string, 0, len(pathParts))
-	normalizedPathTemplateParts := make([]string, 0, len(pathParts))
-	pathLabelParts := make([]string, 0, len(pathParts))
-	for _, p := range pathParts {
-		if p == "" {
-			continue
-		}
-		placeholderName := getVarPathPartName(p)
-		if placeholderName == "" {
-			// this part is not a placeholder: match it exactly
-			matcherPathParts = append(matcherPathParts, p)
-			normalizedPathTemplateParts = append(normalizedPathTemplateParts, p)
-			pathLabelParts = append(pathLabelParts, p)
-
-		} else {
-			// this part is a placeholder: match a wildcard for it
-			matcherPathParts = append(matcherPathParts, regexUrlPathPart)
-			normalizedPathTemplateParts = append(normalizedPathTemplateParts, ":"+placeholderName)
-			pathLabelParts = append(pathLabelParts, "{"+placeholderName+"}")
-		}
-	}
-	rawRegEx := &strings.Builder{}
-	rawRegEx.WriteString("^")
-	rawRegEx.WriteString(regexOptionalSlashes)
-	rawRegEx.WriteString(strings.Join(matcherPathParts, `\/`))
-	rawRegEx.WriteString(regexOptionalSlashes)
-	rawRegEx.WriteString("$")
-
-	matcher = rawRegEx.String()
-	normalizedPathTemplate = strings.Join(normalizedPathTemplateParts, "/")
-	pathLabel = strings.Join(pathLabelParts, "/")
+func generateRegExpStringForPathTemplate(path string) (string, string) {
+	return createPathTemplate(path), createPathPattern(path)
+}
 
-	return
+func createPathTemplate(path string) string {
+	path = regexpMultiSlash.ReplaceAllString(path, "/")
+	path = regexpLeadingSlash.ReplaceAllString(path, "")
+	path = regexpTrailingSlash.ReplaceAllString(path, "")
+	path = rexexpSlashColumnVar.ReplaceAllString(path, "/{$1}")
+	return path
 }
 
-// getVarPathPartName detects if a path template part represents a variadic placeholder.
-// Returns "" when it is not or its name when it is.
-func getVarPathPartName(pathTemplatePart string) string {
-	// check if it is `:id`
-	matches := regexVarPathPartColon.FindStringSubmatch(pathTemplatePart)
-	if len(matches) == 2 {
-		return matches[1]
-	}
-	// check if it is `{id}`
-	matches = regexVarPathPartCurlyBraces.FindStringSubmatch(pathTemplatePart)
-	if len(matches) == 2 {
-		return matches[1]
-	}
-	// it is not a placeholder
-	return ""
+func createPathPattern(path string) string {
+	path = regexpMultiSlash.ReplaceAllString(path, "/")
+	path = regexpLeadingSlash.ReplaceAllString(path, "^/*")
+	path = regexpTrailingSlash.ReplaceAllString(path, "/*$")
+	path = rexexpSlashColumnVar.ReplaceAllString(path, "/.+")
+	path = rexexpCurlyBracketVar.ReplaceAllString(path, ".+")
```

Comment:
Currently, we are missing a step to enable matching of multiple slashes within paths.

Reply:
Done.